public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts

```diff
@@ -55,12 +55,23 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
     this.azureMonitorUrl = `/${managementRoute}/subscriptions`;
 
     this.url = instanceSettings.url || '';
-    this.subscriptionId = this.instanceSettings.jsonData.logAnalyticsSubscriptionId || '';
+    this.defaultSubscriptionId = this.instanceSettings.jsonData.logAnalyticsSubscriptionId || '';
     this.defaultOrFirstWorkspace = this.instanceSettings.jsonData.logAnalyticsDefaultWorkspace || '';
   }
 
   isConfigured(): boolean {
-    return !!this.subscriptionId && this.subscriptionId.length > 0;
+    return !this.validateDatasource();
+  }
+
+  async getSubscriptions(): Promise<Array<{ text: string; value: string }>> {
+    if (!this.isConfigured()) {
+      return [];
+    }
+
+    const url = `${this.azureMonitorUrl}?api-version=2019-03-01`;
```

Comment:
it feels like the `api-version` should be a more visible constant

Reply:
Agree, I do not like that we craft HTTP requests directly in the business logic and use poorly written typings.
This method was copied over from `AzureMonitorDatasource`.

I created a separated work item clean this up:
#34860